answer.js

```diff
@@ -1,17 +1,102 @@
 // Answer here for tasks.md
 
 // Q1 Answer here...
+var mongoose = require('mongoose')
+var Schema = mongoose.Schema;
+var userSchema = new Schema({
+    name: {
+        type: String,
+        require: true,
 
-// Q2 Answer here...
+    },
+    email: {
+        type: String,
+        required: true,
 
+    },
+    age: {
+        type: Number,
+        required: true,
+    }
+})
+// Q2 Answer here...
+userSchema.update({
+    email: {
+        type: String,
+        lowercase: true
+    }
+})
 // Q3 Answer here...
+userSchema.update({
+    password: {
+        type: String,
+        minlength: 5,
+    },
+    createdAt: {
+        type: Date,
+        default: (new Date).getDate()
+    }
+}, )
+module.exports = mongoose.model('User',userSchema);
 
 // Q4 Answer here...
+// 1. village -> String
+//   2. city -> String
+//   3. state -> String
+//   4. pin -> number
+//   5. user -> ObjectId of User from user schem
+var addressSchema = new Schema({
+    village: {
+        type: String,
+        required: true,
+    },
+    city: {
+        type: String,
+        required: true,
+    },
+    state:{
+        type:String,
+        required:true,
+    },
+    pin:{
```

Comment:
you should use type: String for PIN because if it starts with Zero, javascript will omit the leading zero and trim the PIN.
for Example 
```js
var PINCODE = 002345
// It will be saved as simply 2345 which we do't want.
```